pkg/operator/controllers/checker/const.go

```diff
@@ -0,0 +1,6 @@
+package checker
+
+const (
+	credentialNamespace  = "kube-system"
+	credentialSecretName = "azure-credentials"
+)
```

Comment:
I am wondering if using Cluster Credential Operator is more appropriate solution here. Other components in other namespaces (machine-api, registry and ingress) that need access to those credentials do access them from a secret in their respective namespaces, which are reconciled by CCO. 
While the root secret that you are referring to here is expected to be source of truth I have at least 2 concerns about this approach - first it is not consistent with the rest of OpenShift ecosystem and second, at the end of the day we care that the secrets for machine-api, ingress and registry are correct, so if you are using the root secret, but something is broken with CCO, then the root secret maybe correct, but it is not reconciled properly to the components that really need it

Reply:
CCO Credentials request should not be created by running process. They are part of the install payload and should be static. I don't like the fact that we distribute these secrets across the cluster already and this would be one more instance of "copy"

Reply:
@mjudeikis thanks for the input, agree to keep it out of the operator to keep things simpler. 
However nothing in documentation suggests that credentialsRequest should be only used in install and it works on running cluster as well. This is not "one more instance of copy" this is using the same mechanism that OpenShift provides. E.g. machine-api-operator could have also decided to use the root credentials, but they use CCO. 
I guess we can say that ARO operator is a special case, and it uses the secret in a different way than other OpenShift operators.